model/patch_lifecycle.go

```diff
@@ -349,6 +349,10 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		}).TVPairsToVariantTasks()
 	}
 
+	// if variant tasks is still empty, then the patch is empty and we shouldn't add to commit queue
+	if p.Alias == evergreen.CommitQueueAlias && len(p.VariantsTasks) == 0 {
```

Comment:
could this be more general, for any patch?

Reply:
Could theoretically. Wasn't sure if there was a use case for a noop version.